scripts/main/tabindex.js

```diff
@@ -2,92 +2,142 @@
 	// Get all elements which have a tabindex
-	let tmp = $(elem).find("[tabindex]");
+	// TODO @Hallenser: What did you intended by the TODO above? It seems as if the jQuery selector is already as short as possible?
+	const tmp = $(elem).find("[tabindex]");
 
 	// iterate over all elements and set tabindex to stored value (i.e. make is not focussable)
-	tmp.each(function (i, e) {
-		// TODO: shorter notation
-		a = $(e).data("tabindex-saved");
-		$(e).attr("tabindex", a);
-	});
+	tmp.each(
+		/**
+		 * @param {number} i - the index
+		 * @param {Element} e - the HTML element
+		 * @this {Element} - identical to `e`
+		 */
+		function (i, e) {
+			// TODO: shorter notation
+			// TODO @Hallenser: What do you intended by the TODO `short notation`? Moreover: Why do we use `this` and `e`? They refer to the identical instance of a HTML element.
+			const a = $(e).data("tabindex-saved");
+			$(e).attr("tabindex", a);
+		}
+	);
 };
 
+/**
+ * @param {jQuery} elem
+ * @param {boolean} [saveFocusElement=false]
+ * @returns {void}
+ */
 tabindex.makeUnfocusable = function (elem, saveFocusElement = false) {
 	if (!lychee.enable_tabindex) return;
 
-	// Todo: Make shorter noation
+	// Todo: Make shorter notation
 	// Get all elements which have a tabindex
-	let tmp = $(elem).find("[tabindex]");
+	const tmp = elem.find("[tabindex]");
 
 	// iterate over all elements and set tabindex to -1 (i.e. make is not focussable)
-	tmp.each(function (i, e) {
-		$(e).attr("tabindex", "-1");
-		// Save which element had focus before we make it unfocusable
-		if (saveFocusElement && $(e).is(":focus")) {
-			$(e).data("tabindex-focus", true);
-			// Remove focus
-			$(e).blur();
+	tmp.each(
+		/**
+		 * @param {number} i - the index
+		 * @param {Element} e - the HTML element
+		 */
+		function (i, e) {
+			$(e).attr("tabindex", "-1");
+			// Save which element had focus before we make it unfocusable
+			if (saveFocusElement && $(e).is(":focus")) {
+				$(e).data("tabindex-focus", true);
+				// Remove focus
+				$(e).blur();
+			}
 		}
-	});
+	);
 
 	// Disable input fields
-	$(elem).find("input").attr("disabled", "disabled");
+	elem.find("input").attr("disabled", "disabled");
 };
 
+/**
+ * @param {jQuery} elem
+ * @param {boolean} [restoreFocusElement=false]
+ * @returns {void}
+ */
 tabindex.makeFocusable = function (elem, restoreFocusElement = false) {
 	if (!lychee.enable_tabindex) return;
 
-	// Todo: Make shorter noation
+	// Todo: Make shorter notation
 	// Get all elements which have a tabindex
-	let tmp = $(elem).find("[data-tabindex]");
-
-	// iterate over all elements and set tabindex to stored value (i.e. make is not focussable)
-	tmp.each(function (i, e) {
-		$(e).attr("tabindex", $(e).data("tabindex"));
-		// restore focus elemente if wanted
-		if (restoreFocusElement) {
-			if ($(e).data("tabindex-focus") && lychee.active_focus_on_page_load) {
-				$(e).focus();
-				$(e).removeData("tabindex-focus");
+	const tmp = elem.find("[data-tabindex]");
```

Comment:
I know you aren't trying to understand or fix all this code but doesn't this line strike you as strange? Everywhere else (including `tabindex.makeUnfocusable`) we were looking for `"[tabindex]"` yet here it's `"[data-tabindex]"`?!

Reply:
I haven't yet fully understand the code yet. Please note that `.attr('data-tabindex)' ...)` and `.data('tabindex', ...)` are (mostly) the same. The latter is a convenience shortcut for the former. I grepped the code for the regular expression `data.*tabindex` and I also found a lot of matches in `build.json`.

The main problems of this code are its inconsistency, i.e. a mix of `attr` and `data`, no clean separation between this code and `build.js`(that file is awful) and a complete lack of comments what the code is supposed to do and how it interacts with the other modules.

It is not only this code which strakes me as strange.

Reply:
Can we resolve this conversation for now? I assume you don't want to fix `build.js` in the PR, do you?